- add a check to avoid if user does not fill in all fields in form.

iamcxa opened the pull request.

iamcxa committed:
- add a check to avoid if user does not fill in all fields in form.

iamcxa committed:
- change name.length limit from 2 to 1.

smlsunxie commented:
都填了，還是出現提示訊息，確認一下喔

<img width="1079" alt="screen shot 2015-08-28 at 8 08 11 am" src="https://cloud.githubusercontent.com/assets/1689227/9536215/13d58808-4d5c-11e5-94f3-1d611192a59e.png">

iamcxa committed:
Merge remote-tracking branch 'upstream/master'

iamcxa committed:
- change mobile limit from 9 to 8, to fit home phone number.
- add auto transform page from #purchase to #product if postData(cart) is empty.

iamcxa committed:
- add e.stopImmediatePropagation() to prevent submit method runs twice when doing purchase.
- add .unbind('click') to #purchaseButton to prevent show alert twice just in case.

iamcxa committed:
- adject each input's text length limit a bit, ensure it works.

iamcxa committed:
- fix auto transform page to #product if select no product.
- add an alert when user only type 1 word for user/shipment name.
